### ISSUE TITLE
Out of Memory when training an large custom YOLOV3 dataset

### ISSUE THREAD

**Issue by skoroneos** [OlafenwaMoses/ImageAI#609]
While training in a custom large dataset using YOLOV3 i realised that when finishing epoch 1, no memory was released.
Due to the dataset size, it had used more than 50GB of the 64GB memory on the machine on the first epoch and it kept growing while working on Epoch 2.
Seems there is a memory leak or some structure keeps increasing even when epochs change.
This is not an issue with the GPU memory (batch size was 8) as the first epoch trained fine and generated a usable model 
Machine has 64GB of memory plus 8GB of swap and a GTX 1070 with 8GB of memory. All memory and swap were exhausted while no other process was running on the machine (except linux and system processes)

Epoch 1/100
51576/51576 [==============================] - 42746s 829ms/step - loss: 7.3072 - yolo_layer_1_loss: 0.5451 - yolo_layer_2_loss: 2.1719 - yolo_layer_3_loss: 4.5901 - val_loss: 5.1430 - val_yolo_layer_1_loss: 0.4046 - val_yolo_layer_2_loss: 1.7030 - val_yolo_layer_3_loss: 3.9388
Epoch 2/100
13117/51576 [======>.......................] - ETA: 10:02:38 - loss: 5.7555 - yolo_layer_1_loss: 0.3795 - yolo_layer_2_loss: 1.6745 - yolo_layer_3_loss: 3.7015Killed

**Comment by OlafenwaMoses** [comment]
I will advice that you keep the batch size at about 4 to address this.
